lib/kube/proxy/portforward_spdy.go

```diff
@@ -174,77 +173,54 @@ func (h *portForwardProxy) forwardStreamPair(p *httpStreamPair, remotePort int64
 
 	// read and write from the error stream
 	targetErrorStream, err := h.targetConn.CreateStream(headers)
+	h.onPortForward(fmt.Sprintf("%v:%v", h.podName, remotePort), err == nil /* success */)
 	if err != nil {
-		h.onPortForward(fmt.Sprintf("%v:%v", h.podName, remotePort), false)
-		return trace.ConnectionProblem(err, "error creating error stream for port %d", remotePort)
+		err := trace.ConnectionProblem(err, "error creating error stream for port %d", remotePort)
+		p.sendErr(err)
+		p.close()
+		return err
 	}
-	h.onPortForward(fmt.Sprintf("%v:%v", h.podName, remotePort), true)
-	defer targetErrorStream.Close()
-
-	go func() {
-		_, err := io.Copy(targetErrorStream, p.errorStream)
-		if err != nil && err != io.EOF {
-			h.Debugf("Copy stream error: %v.", err)
-		}
+	defer func() {
+		// on stream close, remove the stream from the connection and close it.
+		h.targetConn.RemoveStreams(targetErrorStream)
+		targetErrorStream.Close()
 	}()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
 
-	errClose := make(chan struct{})
 	go func() {
```

Comment:
nit: Keep the `Add` right beside the `go`.

```suggestion

	wg.Add(1)
	go func() {
```